src/controllers/media.ts

```diff
@@ -26,10 +27,24 @@ const MESSAGES = {
  */
 const getFromIMDbAPI = async(imdbId?: string, searchRequest?: SearchRequest): Promise<MediaMetadataInterface> => {
   if (!imdbId) {
-    const searchResults = await imdbAPI.search(searchRequest);
-    // TODO Choose the most appropriate result instead of just the first
-    const searchResult = _.first(searchResults.results);
-    imdbId = searchResult.imdbid;
+    let searchResults;
+    try {
+      searchResults = await imdbAPI.search(searchRequest);
+      // TODO Choose the most appropriate result instead of just the first
+      const searchResult = _.first(searchResults.results);
+      // @ts-ignore
+      imdbId = searchResult.imdbid;
+    } catch (e) {
+      // no movie found, let's try get a TV episode
+      if (e.message.includes('Movie not found')) {
+        const parsedFilename = episodeParser(searchRequest.name);
+        const tvSeriesInfo = await imdbAPI.get({ name: parsedFilename.show });
+        // @ts-ignore
+        const allEpisodes = await tvSeriesInfo.episodes();
```

Comment:
At this point we have all episodes for the series so we could maybe seed the db with all of them, but that would maybe need to be outside of this http request otherwise it could slow things down

Reply:
This looks great and yeah let's make an issue for that. I wonder if we can avoid having the episode search inside the catch here, and instead check whether the filename is for an episode before we query. Do you think that would be good?